src/bootstrap/tool.rs

```diff
@@ -683,6 +683,50 @@ impl Step for LldWrapper {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustAnalyzer {
+    pub compiler: Compiler,
+    pub target: TargetSelection,
+}
+
+impl Step for RustAnalyzer {
+    type Output = Option<PathBuf>;
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
```

Comment:
Why do you have ONLY_HOSTS set? that will prevent people from cross-compiling, which shouldn't be necessary. tests are usually host only but there's nothing target-specific about rust-analyzer that I know of.